libcflib/db.xsh

```diff
@@ -0,0 +1,82 @@
+"""Database for accessing graph information"""
+import os
+
+import zict
+
+from .model import Artifact
+
+class DB:
+    """A logging object for fixie that stores information in line-oriented JSON
+    format.
+    """
+
+    __inst = None
+
+    def __new__(cls):
+        # make the logger a singleton
+        if DB.__inst is None:
+            DB.__inst = object.__new__(cls)
+        return DB.__inst
+
+    def __init__(self, history=100):
+        """
+
+        Parameters
+        ----------
+        history: int
+            Size of the cache
+
+        Caching forked from Streamz
+        Copyright (c) 2017, Continuum Analytics, Inc. and contributors
+        All rights reserved.
+
+        """
+        if os.path.exists($LIBCFGRAPH_DIR):
+            git pull $LIBCFGRAPH_URL master
+        else:
+            git clone $LIBCFGRAPH_URL $LIBCFGRAPH_DIR
+        self.cache_dict = {}
```

Comment:
Probably best not to put type names in variable names.  I would just call this `cache`